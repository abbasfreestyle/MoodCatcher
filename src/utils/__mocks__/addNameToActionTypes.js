export const name = {
  correct: 'MockReducer',
  lowerCase: 'mockReducer',
  invalid: [123, { object: 'object' }, ['array']],
  empty: [undefined, null, false, 0]
};

export const actionTypes = {
  invalid: [['array'], 'string', 123, null, undefined, false],
  mock: {
    MOCK_LOAD: 'MOCK_LOAD',
    MOCK_LOAD_SUCCESS: 'MOCK_LOAD_SUCCESS',
    MOCK_LOAD_ERROR: 'MOCK_LOAD_ERROR'
  },
  expect: {
    MOCK_LOAD: `${name.correct}/MOCK_LOAD`,
    MOCK_LOAD_SUCCESS: `${name.correct}/MOCK_LOAD_SUCCESS`,
    MOCK_LOAD_ERROR: `${name.correct}/MOCK_LOAD_ERROR`
  },
  lowerCases: [
    {
      mock_load: 'MOCK_LOAD'
    },
    { MOCK_LOAD: 'mock_load' },
    { mock_load: 'mock_load' }
  ]
};
