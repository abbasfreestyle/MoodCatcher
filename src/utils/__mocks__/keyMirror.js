export const actionTypes = {
  mock: ['MOCK_LOAD', 'MOCK_LOAD_SUCCESS', 'MOCK_LOAD_ERROR'],
  expected: {
    MOCK_LOAD: 'MOCK_LOAD',
    MOCK_LOAD_SUCCESS: 'MOCK_LOAD_SUCCESS',
    MOCK_LOAD_ERROR: 'MOCK_LOAD_ERROR'
  }
};
