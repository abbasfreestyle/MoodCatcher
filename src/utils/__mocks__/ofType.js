export const types = {
  toTest: [
    { object: 'object' },
    ['array'],
    'some string',
    0,
    undefined,
    false,
    null,
    1234,
    0.234
  ],
  expect: [
    'object',
    'array',
    'string',
    'number',
    'undefined',
    'boolean',
    'null',
    'number',
    'number'
  ]
};
