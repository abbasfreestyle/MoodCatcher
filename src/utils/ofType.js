export const ofType = type => {
  // return the actual primitive type
  switch (true) {
    case type === null:
      // null is a type of object, instead it should return null
      return 'null';
    case Array.isArray(type):
      // an array is also a type of object, it should return array
      return 'array';
    default:
      // The rest is good, it can continue with this
      return typeof type;
  }
};
