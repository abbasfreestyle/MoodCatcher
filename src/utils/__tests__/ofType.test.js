import { types } from '../__mocks__/ofType';
import { ofType } from '../ofType';

describe('Utils > Returning the correct types', () => {
  describe('Snapshots', () => {
    it('oftype mocks should match the snapshot', () => {
      expect(types).toMatchSnapshot();
    });
  });

  describe('Returns the correct value', () => {
    types.toTest.forEach((type, index) => {
      it(`Should return ${JSON.stringify(type)} as type ${
        types.expect[index]
      }`, () => {
        expect(ofType(type)).toEqual(types.expect[index]);
      });
    });
  });
});
