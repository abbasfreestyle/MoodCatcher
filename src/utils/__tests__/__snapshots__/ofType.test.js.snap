// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils > Returning the correct types Snapshots oftype mocks should match the snapshot 1`] = `
Object {
  "expect": Array [
    "object",
    "array",
    "string",
    "number",
    "undefined",
    "boolean",
    "null",
    "number",
    "number",
  ],
  "toTest": Array [
    Object {
      "object": "object",
    },
    Array [
      "array",
    ],
    "some string",
    0,
    undefined,
    false,
    null,
    1234,
    0.234,
  ],
}
`;
